netstat/netstat_linux.go

```diff
@@ -250,11 +250,11 @@ func extractProcInfo(sktab []SockTabEntry) {
 // doNetstat - collect information about network port status
 func doNetstat(path string, fn AcceptFn) ([]SockTabEntry, error) {
 	f, err := os.Open(path)
+	defer f.Close()
 	if err != nil {
 		return nil, err
 	}
 	tabs, err := parseSocktab(f, fn)
-	f.Close()
 	if err != nil {
```

Comment:
Sorry, but, you're missing the point that you should only close file descriptors that
opened successfully.